src/Pay/Adapter.php

```diff
@@ -72,43 +72,40 @@ public function getCurrency(): string
     /**
      * Make a purchase request
      */
-    abstract public function purchase(int $amount, string $customerId, string $cardId, array $additionalParams = []): array;
+    abstract public function purchase(int $amount, string $customerId, string $paymentMethodId, array $additionalParams = []): array;
 
     /**
      * Refund payment
      */
-    abstract public function refund(string $paymentId, int $amount = null): array;
+    abstract public function refund(string $paymentId, int $amount = null, string $reason = null): array;
 
     /**
-     * Add a credit card for a customer
+     * Add a payment method
      */
-    abstract public function createCard(string $customerId, string $cardId): array;
+    abstract public function createPaymentMethod(string $customerId, string $type, array $details): array;
```

Comment:
@lohanidamodar can't we just keep the same signature for the abstraction and change the internal API?

Reply:
I updated this as I felt this is more suitable, and also better later to support different types of payment methods other than card?

Reply:
Should we add constants for $type?

Reply:
We can, right now only support card

Reply:
Sounds good.

Reply:
Also this function is only used for testing as we don't create the payment method ourselves, it's handled by Stripe's JS elements so that we don't have to know and save the card information